src/jobsautoreport/report.py

```diff
@@ -1,31 +1,188 @@
+    def _is_rehearsal(job: JobDetails) -> bool:
+        return (
+            "rehearse" in job.name
+            and job.type == "presubmit"
+            and job.refs.repo == "release"
+            and job.refs.org == "openshift"
+        )
+
+    @staticmethod
+    def _is_assisted_repository(job: JobDetails) -> bool:
+        return (
+            job.refs.repo
+            in [
+                "assisted-service",
+                "assisted-installer",
+                "assisted-installer-agent",
+                "assisted-image-service",
+                "assisted-test-infra",
+                "cluster-api-provider-agent",
+            ]
+            and job.refs.org == "openshift"
+        )
+
+    @staticmethod
+    def _is_e2e_or_subsystem_class(job: JobDetails) -> bool:
+        return "e2e" in job.name or "subsystem" in job.name
+
+    def get_report(self, from_date: datetime, to_date: datetime) -> Report:
+        jobs = self._querier.query_all_jobs(from_date, to_date)
+        rehearsal_jobs = [job for job in jobs if self._is_rehearsal(job)]
+        subsystem_and_e2e_jobs = [
+            job
+            for job in jobs
+            if self._is_assisted_repository(job)
+            and self._is_e2e_or_subsystem_class(job)
+        ]
+        periodic_subsystem_and_e2e_jobs = [
+            job for job in subsystem_and_e2e_jobs if job.type == JobType.PERIODIC.value
+        ]
+        presubmit_subsystem_and_e2e_jobs = [
+            job for job in subsystem_and_e2e_jobs if job.type == JobType.PRESUBMIT.value
+        ]
+
+        return Report(
+            number_of_e2e_or_subsystem_periodic_jobs=len(
+                periodic_subsystem_and_e2e_jobs
+            ),
+            number_of_successful_e2e_or_subsystem_periodic_jobs=self._get_number_of_jobs_by_state(
+                periodic_subsystem_and_e2e_jobs, JobState.SUCCESS
+            ),
+            number_of_failing_e2e_or_subsystem_periodic_jobs=self._get_number_of_jobs_by_state(
+                periodic_subsystem_and_e2e_jobs, JobState.FAILURE
+            ),
+            success_rate_for_e2e_or_subsystem_periodic_jobs=self._get_success_rate(
+                periodic_subsystem_and_e2e_jobs
+            ),
+            top_10_failing_e2e_or_subsystem_periodic_jobs=self._get_top_n_jobs(
+                periodic_subsystem_and_e2e_jobs, self._get_job_success_rate, 10, False
+            ),
+            number_of_e2e_or_subsystem_presubmit_jobs=len(
+                presubmit_subsystem_and_e2e_jobs
+            ),
+            number_of_successful_e2e_or_subsystem_presubmit_jobs=self._get_number_of_jobs_by_state(
+                presubmit_subsystem_and_e2e_jobs, JobState.SUCCESS
+            ),
+            number_of_failing_e2e_or_subsystem_presubmit_jobs=self._get_number_of_jobs_by_state(
+                presubmit_subsystem_and_e2e_jobs, JobState.FAILURE
+            ),
+            number_of_rehearsal_jobs=self._get_number_of_rehearsal_jobs(rehearsal_jobs),
```

Comment:
```suggestion
            number_of_rehearsal_jobs=len(rehearsal_jobs),
```
and remove of  _get_number_of_rehearsal_jobs